cli/src/main/java/org/jboss/as/cli/handlers/DeploymentHandler.java

```diff
@@ -117,8 +131,60 @@ protected ModelNode buildReadDeploymentResourceRequest(String name) {
         return request;
     }
 
-    protected MountHandle extractArchive(File archive, TempFileProvider tempFileProvider) throws IOException {
-        return ((MountHandle)VFS.mountZipExpanded(archive, VFS.getChild("cli"), tempFileProvider));
+    static File extractArchive(final File cliArchive) throws IOException {
+        final ZipInputStream zis = new ZipInputStream(new FileInputStream(cliArchive));
```

Comment:
this will produce file handle leak.
moving this to try-with-resources block would eliminate the leak as well.